janus_core/src/test_util/kubernetes.rs

```diff
@@ -2,21 +2,92 @@
 
 use kube::config::{KubeConfigOptions, Kubeconfig};
 use rand::{thread_rng, Rng};
-use std::process::{Command, Stdio};
-use tempfile::{NamedTempFile, TempPath};
+use std::{
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+};
+use tempfile::NamedTempFile;
+use tracing::debug;
+
+/// Cluster represents a running Kubernetes cluster.
+pub struct Cluster {
+    /// Path to the Kubernetes config file, e.g., `~/.kube/config`
+    kubeconfig_path: PathBuf,
+    /// Name of the `kubectl` context to use.
+    context_name: String,
+}
+
+impl Cluster {
+    /// Creates a handle to an existing Kubernetes cluster.
+    pub fn create<P: AsRef<Path>>(kubeconfig_path: P, context_name: &str) -> Self {
+        let kubeconfig_path = kubeconfig_path.as_ref();
+        debug!(?kubeconfig_path, context_name, "Creating cluster handle");
+        Self {
+            kubeconfig_path: kubeconfig_path.to_path_buf(),
+            context_name: context_name.to_owned(),
+        }
+    }
+
+    /// Returns a new [`kube::Client`] configured to interact with this Kubernetes cluster.
+    pub async fn client(&self) -> kube::Client {
+        kube::Client::try_from(
+            kube::Config::from_custom_kubeconfig(
+                Kubeconfig::read_from(&self.kubeconfig_path).unwrap(),
+                &KubeConfigOptions {
+                    context: Some(self.context_name.clone()),
+                    ..KubeConfigOptions::default()
+                },
+            )
+            .await
+            .unwrap(),
+        )
+        .unwrap()
+    }
+
+    /// Set up port forwarding from `local_port` to `service_port` on the service in the namespace.
+    /// Returns a [`PortForwardGuard`] which will kill the child `kubectl` process on drop.
+    pub fn forward_port(
+        &self,
+        namespace: &str,
+        service: &str,
+        local_port: u16,
+        service_port: u16,
+    ) -> PortForwardGuard {
+        let args = [
+            "--kubeconfig",
+            &self.kubeconfig_path.to_string_lossy(),
+            "--context",
+            &self.context_name,
+            "--namespace",
+            namespace,
+            "port-forward",
+            &format!("service/{service}"),
+            &format!("{local_port}:{service_port}"),
+        ];
+        debug!(?args, "Invoking kubectl");
+        let child = Command::new("kubectl")
+            .args(args)
+            .stdin(Stdio::null())
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .unwrap();
+
+        PortForwardGuard { child }
+    }
+}
 
 /// EphemeralCluster represents a running ephemeral Kubernetes cluster for testing. Dropping an
 /// EphemeralCluster will cause the associated Kubernetes cluster to be stopped & cleaned up.
 pub struct EphemeralCluster {
-    name: String,
-    kubeconfig_path: TempPath,
+    pub cluster: Cluster,
```

Comment:
rather than making this `pub`, I suggest adding a getter that returns a reference to the `&Cluster`.